scripts/Muon/GUI/ElementalAnalysis2/context/ea_group_context.py

```diff
@@ -118,3 +120,46 @@ def remove_group(self, group_name):
             if group.name == group_name:
                 self._groups.remove(group)
                 return
+
+    @staticmethod
+    def get_detector_and_run_from_workspace_name(workspace_name):
+        for suffix in INVALID_STRINGS_FOR_GROUP_NAMES:
+            if suffix in workspace_name:
+                workspace_name = workspace_name.replace(suffix, "")
+
+        run, detector = workspace_name.split(";")
+
+        return run.strip(), detector.strip()
+
+    def remove_workspace_from_group(self, workspace_name):
+        """
+        handles removing workspace from group
+        :param workspace_name : name of workspace removed
+        """
+        if workspace_name.endswith(REBINNED_FIXED_WS_SUFFIX):
```

Comment:
you can make this a method that returns group and then loop over a list of strings:

```
for name in [ REBIN_FIXED, PEAKS]:
    group = self.check_for_str(name)
    if group is None:
       return
    if name == REBIN_FIXED:
       ....
```

Reply:
also is it possible to get peaks with rebinned etc?

Reply:
No the findpeakalgorithm is only run on the raw data

Reply:
could we use some else/elif?